File: src/app/models/Book.ts
export class Book {
    
    id: number;
    title: string;
    price: number;
    author: string;

    constructor(title: string, price: number, author: string, id=null) {
        this.title = title;
        this.price = price;
        this.author = author;
        if (id != null) this.id = id
    }  
}

File: book-web-api/AspNetCoreWebApi-Assessment/Models/Book.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AspNetCoreWebApi_Assessment
{
    public class Book
    {
        private int id;
        private string title;
        private decimal? price;
        private string author;
        public Book() {}

        public Book(string title, decimal? price, string author)
        {
            this.title = title;
            this.price = price;
            this.author = author;
        }

        public Book(int id, string title, decimal? price, string author)
        {
            this.id = id;
            this.title = title;
            this.price = price;
            this.author = author;
        }

        public int Id { get => id; set => id = value; }
        public string Title { get => title; set => title = value; }
        public decimal? Price { get => price; set => price = value; }
        public string Author { get => author; set => author = value; }
    }
}
